### ISSUE TITLE
Fatal error on Product-Compare page when Short Description & Description attributes are empty

### ISSUE THREAD

**Issue by operator888** [magento/magento2#28254]
![magento2-3-5-dev-bug t23dev-2020 05 17-21-40-01 (1)](https://user-images.githubusercontent.com/3840384/82198670-ea66af00-98fc-11ea-8057-f50126ef26a4.jpg)

### Preconditions (*)

1. Magento 2.3.5

### Steps to reproduce (*)

1.  Add products to Compare where at least one product has no value set for at least one of its comparable attributes (e.g. "Description" or "Short Description" is empty).
2. Go to the compare page (<site_root>/catalog/product_compare/index)

### Expected result (*)

1. Compare page should load normally, where any unconfigured attributes show "N/A" or "No".

### Actual result (*)

1. Compare page is Broken and a Fatal Error is displayed. 
Fatal error: Uncaught TypeError: preg_match() expects parameter 2 to be string, object given in /app/web/vendor/magento/module-catalog/Helper/Output.php:245 

`Fatal error: Uncaught TypeError: preg_match() expects parameter 2 to be string, object given in /app/web/vendor/magento/module-catalog/Helper/Output.php:245 Stack trace: #0 /app/web/vendor/magento/module-catalog/Helper/Output.php(245): preg_match('/{{([a-z]{0,10}...', Object(Magento\Framework\Phrase)) #1 /app/web/vendor/magento/module-catalog/Helper/Output.php(185): Magento\Catalog\Helper\Output->isDirectivesExists(Object(Magento\Framework\Phrase)) #2 /app/web/vendor/magento/module-catalog/view/frontend/templates/product/compare/list.phtml(134): Magento\Catalog\Helper\Output->productAttribute(Object(Magento\Catalog\Model\Product\Interceptor), Object(Magento\Framework\Phrase), 'short_descripti...') #3 /app/web/vendor/magento/framework/View/TemplateEngine/Php.php(59): include('/app/web/vendor...') #4 /app/web/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\View\TemplateEngine\Php->render(Object(Magento\Catalog\Block\Product\Compare\ListCompare\Interceptor), '/app/web/vendor...', Array) #5 /app/w in /app/web/vendor/magento/module-catalog/Helper/Output.php on line 245`

**Comment by Brengineer** [comment]
(Note that this is related to [Issue #20229](https://github.com/magento/magento2/issues/20229), which was not fixed properly.)

**Comment by operator888** [comment]
> (Note that this is essentially the same issue as [#20229](https://github.com/magento/magento2/issues/20229), which was not fixed properly.)

Yes Brengineer, as you can see on the bottom of #20229 report, I have made an entry yesterday, but noone seem to re-open that case or take any action. So, I had to re-report it here, because although its about the same topic, its still not fixed, and I have no Idea why that report had been closed.

**Comment by Brengineer** [comment]
> Yes Brengineer, as you can see on the bottom of #20229 report, I have made an entry yesterday, but noone seem to re-open that case or take any action. So, I had to re-report it here, because although its about the same topic, its still not fixed, and I have no Idea why that report had been closed.

I agree. I just wanted to cross-reference. 😉

**Comment by operator888** [comment]
But I cross-reference and clearly mention that relation with that bug-report 2 messages before your message in here.

<img width="1291" alt="Screenshot 2020-05-19 at 10 01 36" src="https://user-images.githubusercontent.com/3840384/82300979-e350a700-99b7-11ea-8552-ef93b5e9a674.png">

**Comment by Brengineer** [comment]
I'm sorry, I did not notice that entry.
If you like, I will remove my comments.

**Comment by operator888** [comment]
> I'm sorry, I did not notice that entry.
> If you like, I will remove my comments.

its ok with me friend and no problem. 
I just try to be thorough and right when doing bug reports and i was like, huh.. but i reported that already what is he talking about?! :) 
Now lets quit small talk, and work on this and other issues. 
Have a nice day!

**Comment by bernd-reindl** [comment]
Hi.

Should be easy to solve. Add (string) before $attributeHtml. So Phrase is converted to string in preg_match().
```
    public function isDirectivesExists($attributeHtml)
.....
            if (preg_match($pattern, (string)$attributeHtml)) {
.....
    }
```

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @sudheers-kensium 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-34464` were created

**Issue Available**: @sudheers-kensium, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by CarmineDamore78** [comment]
@magento give me 2.3.5 instance

**Comment by magento-engcom-team** [comment]
Hi @CarmineDamore78. Thank you for your request. I'm working on Magento 2.3.5 instance for you

**Comment by magento-engcom-team** [comment]
Hi @CarmineDamore78, here is your [Magento instance](https://i-28254-2-3-5.instances.magento-community.engineering).
Admin access: [https://i-28254-2-3-5.instances.magento-community.engineering/admin_b603](https://i-28254-2-3-5.instances.magento-community.engineering/admin_b603)
Login: `55914a98` Password: `7dc37a2889a8`
Instance will be terminated in up to 3 hours.

**Comment by CarmineDamore78** [comment]
@magento I reproduced the error on M2.3.5

**Comment by CarmineDamore78** [comment]
@magento assign to me this

**Comment by CarmineDamore78** [comment]
@magento I am working on this

**Comment by CarmineDamore78** [comment]
@operator888  
This issue is already fixed on M2.4-develop
Unfortunately we can not backlog our fixes on M.2.3.5
here the patch to apply on M2.3.5
https://github.com/CarmineDamore78/magento2/blob/bugfix-28254_fix_compare_page/0001-bugfix-28254-Fixed-issue-for-compare-page.patch

**Comment by CarmineDamore78** [comment]
@magento fixed on M2.3.5

**Comment by operator888** [comment]
> @magento fixed on M2.3.5

Thank you for your help and your time @CarmineDamore78 !

**Comment by vrajesh-patel1993** [comment]
@magento I am working on this

**Comment by sdzhepa** [comment]
Hello @operator888

Just checked and the issue has been fixed and available in 2.3.5-p2 and `2.4.0`
